### ISSUE TITLE
C# Code coloring glitches after method that returns Task of tuple

### ISSUE THREAD

**Issue by aeschli** [dotnet/csharp-tmLanguage#97]
_From @ahmadalli on December 23, 2017 13:25_

- VSCode Version: Code 1.19.1 (0759f77bb8d86658bc935a10a64f6182c5a1eeba, 2017-12-19T09:35:14.671Z)
- OS Version: Darwin x64 17.3.0
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
swagger-viewer|Arj|1.5.0
csharp|ms-|1.13.1

---

Steps to Reproduce:

1. Have a c# code with a method that returns task of a tuple.
2. this happens (take a look at the coloring of lines 73 to 98):

![image](https://user-images.githubusercontent.com/2167301/34319878-f05904e8-e801-11e7-9d2b-49b208a3c53d.png)

Reproduces without extensions: Yes (the image above is taken with `--disable-extensions` flag)

_Copied from original issue: Microsoft/vscode#40754_

**Comment by aeschli** [comment]
_From @weinand on December 23, 2017 14:0_

Please file C# issues against the C# extension.

**Comment by aeschli** [comment]
_From @ahmadalli on December 23, 2017 14:2_

I disabled the c# extension. this is just code coloring.

**Comment by damieng** [comment]
I don't think this is a problem with Task of Tuple but rather a grammar-wide issue with splitting certain constructs across multiple lines as covered in #62 and #63

**Comment by damieng** [comment]
I don't think this is a problem with Task of Tuple but rather a grammar-wide issue with splitting certain constructs across multiple lines as covered in #62 and #63

**Comment by damieng** [comment]
I don't think this is a problem with Task of Tuple but rather a grammar-wide issue with splitting certain constructs across multiple lines as covered in #62 and #63

**Comment by damieng** [comment]
I don't think this is a problem with Task of Tuple but rather a grammar-wide issue with splitting certain constructs across multiple lines as covered in #62 and #63

**Comment by damieng** [comment]
I don't think this is a problem with Task of Tuple but rather a grammar-wide issue with splitting certain constructs across multiple lines as covered in #62 and #63
